benchmarks/targets/gaussian.jl

```diff
@@ -0,0 +1,54 @@
+using InteractiveUtils: @code_warntype
+
+function ℓπ_gaussian_impl(m, s, x::AbstractVecOrMat{T}) where {T}
```

Comment:
You might want to consider re-writing this as follows:
```julia

function new_impl(m, s, x::AbstractVecOrMat{T}) where {T}
    return .-(log(2 * T(pi)) .+ 2 .* log.(s) .+ abs2.(x .- m) ./ s.^2) ./ 2
end

using BenchmarkTools
m = 0.0
s = 1.0
x = randn(10)

@benchmark ℓπ_gaussian_impl($m, $s, $x)
BenchmarkTools.Trial:
  memory estimate:  640 bytes
  allocs estimate:  4
  --------------
  minimum time:     244.509 ns (0.00% GC)
  median time:      288.242 ns (0.00% GC)
  mean time:        368.230 ns (19.22% GC)
  maximum time:     21.442 μs (97.24% GC)
  --------------
  samples:          10000
  evals/sample:     401

@benchmark new_impl($m, $s, $x)
BenchmarkTools.Trial:
  memory estimate:  160 bytes
  allocs estimate:  1
  --------------
  minimum time:     117.426 ns (0.00% GC)
  median time:      123.827 ns (0.00% GC)
  mean time:        145.419 ns (11.69% GC)
  maximum time:     7.526 μs (97.70% GC)
  --------------
  samples:          10000
  evals/sample:     916

```

Reply:
Thanks, Will. It does help the benchmark! I'm doing it in my own repo and will share the results later with you.